templates/README.md

```diff
@@ -0,0 +1,26 @@
+# How to write a template
+
+Create a new directory alongside this README. The name of the directory is the
+name of the template. Within that directory, create a `template.yaml` that
+follows this syntax:
+
+```yaml
+# Keys in the root of the yaml specify the supported bases
+core18:
+  apps:
+    '*':
+      # YAML snippet you want to add to apps using this template (e.g. `plugs`)
+
+  parts:
+    '*':
+      # Yaml snippet you want to add to parts in the `snapcraft.yaml` (e.g.
+      # `after`)
+
+    # You can also put whatever parts here you want, and they will be put
+    # alongside the other parts in the `snapcraft.yaml`. Take note of the
+    # `$SNAPCRAFT_TEMPLATES_DIR` variable that is pointing to the directory
+    # containing this README.
+    my-template-part:
+      plugin: dump
+      source: $SNAPCRAFT_TEMPLATES_DIR/my-template-name/src
```

Comment:
ok, this might be the reason. Maybe this should be more encapsulated to the actual template instead?

Reply:
Well, remember that templates can only alter the `snapcraft.yaml`. They're not particularly special beasts, we can't really do any magic there or we kill the ability for someone to copy/paste them into their local `snapcraft.yaml` to tweak.

Reply:
oh, I mean SNAPCRAFT_TEMPLATE_DIR specific to the template and not the entire namespace